fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/session/SessionFactory.java

```diff
@@ -89,12 +81,12 @@ public void init() {
     }
 
     /**
-     * Get a new JCR Session
+     * Get a new fedora transaction
      *
-     * @return an internal session
+     * @return an fedora transaction
      */
-    public FedoraSession getInternalSession() {
-        return repo.login();
+    public FedoraTransaction getNewTransaction() { // Should this be read-only?
```

Comment:
getNewTransaction - Added this method as a replacement for the getInternalSession which was used by the WebAcFilter, WebacRolesProvider, and LinkHeaderProvider to read the resource. As the transaction is not directly related to HttpRequest and it is only for reading, we can change this to getNewReadOnlyTransaction and make the transaction read-only.

Reply:
Do we need to track read only transactions at all?  Could the WebAC layer use the PersistentStorageSessionFactory.getReadOnlySession() directly (or some kernel layer provider) instead?

Reply:
Decision from meeting: Jared is working on a FedoraResourceFactory? kernel class and the system internal read-only operations should be routed through that instead of using transactions.

Reply:
`SessionFactory` is handling out transactions? Wouldn't that be the job of `TransactionManagerImpl`? I missed the discussion, so if someone could explain the flow I would appreciate it.